modules/azure/loadbalancer.go

```diff
@@ -0,0 +1,201 @@
+	require.NoError(t, err)
+	return ipAddress, ipType
+}
+
+// GetLoadBalancerFrontendConfigE gets the IP address and specifies public or private for the specified Load Balancer.
+func GetLoadBalancerFrontendConfigE(feConfigName string, loadBalancer01Name string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType, err1 error) {
+	// Get the specified Load Balancer Frontend Config
+	feConfig, err := GetLoadBalancerFrontendIPConfigurationE(feConfigName, loadBalancer01Name, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", -1, err
+	}
+
+	// Get the Properties of the Frontend Configuration
+	feProps := *feConfig.FrontendIPConfigurationPropertiesFormat
+
+	// Check Public Type
+	if feProps.PublicIPAddress != nil {
+		// Get PublicIPAddressResource name for Load Balancer Frontend Configuration
+		pipName := GetNameFromResourceID(*feProps.PublicIPAddress.ID)
+
+		ipValue, err := GetIPOfPublicIPAddressByNameE(pipName, resourceGroupName, subscriptionID)
+		if err != nil {
+			return "", -1, err
+		}
+
+		// Public IP
+		return ipValue, IPType(PublicIP), nil
+	}
+
+	// Private IP
+	return *feProps.PrivateIPAddress, IPType(PrivateIP), nil
+
+}
+
+// GetLoadBalancerFrontendIPConfigurationE gets a Load Balancer Configuration in the specified Azure Resource Group
+func GetLoadBalancerFrontendIPConfigurationE(feConfigName string, loadBalancerName string, resourceGroupName string, subscriptionID string) (*network.FrontendIPConfiguration, error) {
+	// Validate Azure Resource Group Name
+	resourceGroupName, err := getTargetAzureResourceGroupName(resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the client reference
+	client, err := GetLoadBalancerFrontendIPConfigurationClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the Load Balancer Frontend Configuration
+	lbc, err := client.Get(context.Background(), resourceGroupName, loadBalancerName, feConfigName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &lbc, nil
+}
+
+// GetLoadBalancerFrontendIPConfigurationClientE creates a Load Balancer Configuration client.
+func GetLoadBalancerFrontendIPConfigurationClientE(subscriptionID string) (*network.LoadBalancerFrontendIPConfigurationsClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the Load Balancer Frontend Configuration client
+	client := network.NewLoadBalancerFrontendIPConfigurationsClient(subscriptionID)
+
+	// Create an authorizer
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+	client.Authorizer = *authorizer
+
+	return &client, nil
+}
+
+// GetLoadBalancerE gets a Load Balancer in the specified Azure Resource Group
+func GetLoadBalancerE(loadBalancerName string, resourceGroupName string, subscriptionID string) (*network.LoadBalancer, error) {
```

Comment:
Is there any reason not to have a GetLoadBalancer wrapper to expose whole load balancer?

Reply:
The tests do not call the LoadBalancer directly so left the resource call out. The module uses the E version internally. Please advise and can add it.

Reply:
Added